src/storage-client/src/controller.rs

```diff
@@ -2150,7 +2150,7 @@ mod persist_write_handles {
                                                         // by a failure on our next interaction with the catalog stash,
                                                         // but we would incorrectly think this committed and may serve
                                                         // results in the meantime.
-                                                        result = Ok(Ok(Ok(())))
+                                                        result = Ok(Ok(()))
```

Comment:
I think there’s more that can be removed here, no? The stanzas above appear to be attempting to retry indeterminate errors.

Reply:
Yeah, I think effectively this whole loop can go away. (And probably should -- the while loop is now retrying on invalid-usage errors, which we don't want!)

Reply:
great catch! done